iree/compiler/Dialect/VM/Conversion/UtilToVM/test/alignment_ops.mlir

```diff
@@ -0,0 +1,51 @@
+// RUN: iree-opt -split-input-file -iree-vm-conversion -cse %s | IreeFileCheck %s
+
+// CHECK-LABEL: @utilAlign
+func @utilAlign(%arg0 : index, %arg1: index) ->  (index) {
+  %result = util.align %arg0, %arg1 : index
+  //CHECK-DAG: %c1 = vm.const.i32 1 : i32
+  //CHECK-DAG: %0 = vm.sub.i32 %arg1, %c1 : i32
+  //CHECK-DAG: %1 = vm.add.i32 %arg0, %0 : i32
+  //CHECK-DAG: %2 = vm.not.i32 %0 : i32
+  //CHECK-DAG: %3 = vm.and.i32 %1, %2 : i32
+
+  //CHECK-DAG: vm.return %3 : i32
+  return %result : index
+}
+
+// -----
+
+// CHECK-LABEL: @packDynamic
+func @packDynamic(%arg1: index, %arg2: index) -> (index, index, index, index) {
```

Comment:
This is a bit more of a regression test for `iree/compiler/Dialect/HAL/Transforms/PackAllocations.cpp` than a unit test for `iree/compiler/Dialect/VM/Conversion/UtilToVM/ConvertAlignmentOps.cpp`

Can you distill something specific to test here that isn't covered by the test case above? When canonicalizations / folding are added, those tests would fit more naturally here.

Reply:
I get your point. I wanted to continue the test case from the original pack_allocations.mlir thats now essentially split with `Util.AlignOp`. I could just remove this since it doesn't add anything more beyond the test case above. 

If there are no other thoughts where this should go, will remove.

Reply:
Scott suggestions: Check for branches? If its not a power of 2 (I don't think there is anything that checks). Also check for i32 and i64.